test cases/common/262 configure file output format/meson.build

```diff
@@ -0,0 +1,33 @@
+project('configure file output format')
+
+data = configuration_data()
+data.set_quoted('str', 'hello world!', description: '''This is
+a multiline
+description''')
+data.set('int', 42, description: 'ultimate question of life, the universe, and everything')
+data.set('bool', true)
+data.set('false', false)
+
+config_h = configure_file(
+    configuration: data,
+    output_format: 'c',
+    output: 'config.h'
+)
+
+config_nasm = configure_file(
+    configuration: data,
+    output_format: 'nasm',
+    output: 'config.nasm'
+)
+
+config_json = configure_file(
+    configuration: data,
+    output_format: 'json',
+    output: 'config.json'
+)
+
+py = find_program('python3')
+
+test('c_output', py, args: [meson.source_root() / 'compare.py', meson.source_root() / 'expected/config.h', config_h])
+test('nasm_output', py, args: [meson.source_root() / 'compare.py', meson.source_root() / 'expected/config.nasm', config_nasm])
+test('json_output', py, args: [meson.source_root() / 'compare.py', meson.source_root() / 'expected/config.json', config_json])
```

Comment:
You can use `files('compare.py')` etc here instead of source_root() which is deprecated anyways.